lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -178,12 +185,12 @@ static bool validateCodingKeysEnum(DerivedConformance &derived,
       case DoesNotConform:
         it->second->diagnose(diag::codable_non_conforming_property_here,
                              derived.getProtocolType(), it->second->getType());
-        LLVM_FALLTHROUGH;
+        propertiesAreValid = false;
+        break;
 
       case TypeNotValidated:
         // We don't produce a diagnostic for a type which failed to validate.
         // This will produce a diagnostic elsewhere anyway.
-        propertiesAreValid = false;
```

Comment:
Sometimes we *do* want to fail here to trigger a conformance error,
``` swift
struct Foo: Codable {
  enum CodingKeys: CodingKey {
    case x
    case y
  }
  var x: Int
  var y: Undefined
}
```
so I think we should keep this and skip redeclarations – which do not contribute to the conformance – while populating the `properties` map instead (right now we always end up validating the redeclaration rather than original).